data/src/error_analysis/erroranalysis.py

```diff
@@ -0,0 +1,70 @@
+"""Error Analysis tool for G2P.
+
+This script assumes 2 input files. a) Covering Grammer b) Test output. 
+CG file  : Each line contain graphime and their corresponding pronunciation seperated by a tab
+Test File: contains three attributes in each line seperated by a tab. The attributes are - orthography, expected pronunciation, and hypothesized pronunciation."""
+
+__author__ = "Arundhati Sengupta"
+
+import pynini
+import argparse
+
+def main(args: argparse.Namespace):
+    cg_fst = pynini.string_file(args.cg_path).closure().optimize()
+    rulematch_predMatch = 0
+    rulematch_predNotMatch = 0
+    notRulematch_predMatch = 0
+    notRulematch_predNotMatch = 0
+    total_records = 0
+    with open(args.test_path, "r") as source:
+        for line in source:
+            total_records+=1
+            parts = line.split('\t')
+            lg = parts[0].strip()
+            act = parts[1].replace(' ','').replace('.','').strip()
+            predPron = parts[2].replace(' ','').replace('.','').strip()
+             
+            lattice = (lg @ cg_fst @ predPron).project(True)
```

Comment:
This won't work on recent Pynini (2.1.2 or better): please upgrade.

Also this is a good case for using the `rewrite` library.

Reply:
I am working on it. I need to update my Pynini version and thereafter update the code.

Reply:
Sounds good, take your time.